actions/githookHandler.go

```diff
@@ -0,0 +1,87 @@
+package actions
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventgrid/2018-01-01/eventgrid"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/gobuffalo/envy"
+	"github.com/google/go-github/github"
+	"github.com/satori/go.uuid"
+)
+
+// GithookReceiver converts a Github event into an Eventgrid event and publishes the event to Eventgrid
+func GithookReceiver(c buffalo.Context) error {
+	webhookSecret := envy.Get("webhookSecret", "")
+	eventgridSecret := envy.Get("eventgridSecret", "")
+	eventgridEndpoint := envy.Get("eventgridEndpoint", "")
+
+	request := c.Request()
+
+	// save request body so it can be read multiple times
+	var bodyBytes []byte
+
+	if request.Body != nil {
+		// max expected body is 25 megabytes
+		var thirtyMegaBytes int64 = 30000000
+		limitReader := io.LimitReader(request.Body, thirtyMegaBytes)
+		bodyBytes, _ = ioutil.ReadAll(limitReader)
+	}
+
+	request.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+	// validate githook using secret
+	_, err := github.ValidatePayload(request, []byte(webhookSecret))
+
+	if err != nil {
+		c.Logger().Errorf("Validation failed: err=%s\n", err)
+		return c.Error(http.StatusInternalServerError, err)
+	}
+
+	// parse webhook body
+	defer request.Body.Close()
```

Comment:
This line needs to get moved up above where `request.Body` is reassigned to a NopCloser.